Update metrics presented on import flow

renatho opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

## Proposed Changes

* Overhaul of the metrics presented on the import flow, presenting more real and relevant numbers and copy.

Discussion and more context: p1715972644965929-slack-C02NWDZBL0H

#### Default items

<img width="347" alt="Screenshot 2024-05-24 at 12 31 15" src="https://github.com/Automattic/wp-calypso/assets/876340/38e0166e-37ca-4855-9df2-aacd73fd3009">

#### Dynamic items

<img width="348" alt="Screenshot 2024-05-24 at 12 32 10" src="https://github.com/Automattic/wp-calypso/assets/876340/165874c4-511d-421b-9a83-e6855a90667e">

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Some of the previous numbers and copy weren't very accurate.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout branch or use the live link.
* Navigate to: `/setup/site-migration/site-migration-upgrade-plan?siteSlug={FREE_SITE_SLUG}&from=https://unknown.unknown&option=everything`.
* See the defaults, and check if it makes sense.
* Manually update the line `siteMetricData?.basic?.fid?.value` in the codebase to `150` (a not good FID).
* Navigate to: `/setup/site-migration/site-migration-upgrade-plan?siteSlug={FREE_SITE_SLUG}&from=https://www.mazdausa.com&option=everything`.
* Check that the dynamic information was displayed and it makes sense.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

renatho committed:
Update default metrics fallback

renatho commented on client/blocks/importer/wordpress/upgrade-plan/constants.tsx:
```diff
@@ -49,23 +49,26 @@ export const UpgradePlanHostingTestimonials = [
-		id: 'increased-speed',
-		title: translate( 'Increased speed' ),
-		description: translate( '30% faster' ),
+export const defaultHostingDetails: {
```
Refactor it from array to object, as suggested in https://github.com/Automattic/wp-calypso/pull/90717#discussion_r1603326181.

renatho committed:
Update boost text

renatho commented on client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-hosting-details.tsx:
```diff
@@ -77,11 +77,16 @@ export const UpgradePlanHostingDetails = () => {
 					<p className="import__upgrade-plan-hosting-details-header-subtext">
-						{ translate( 'Check our performance, compared to the average WordPress host' ) }
+						{ translate(
+							'We offer a proven 25% boost* in Core Web Vitals—endorsed by data from Google.'
```
See the base for this calc: pdxWSz-1pk-p2

renatho commented on client/blocks/importer/wordpress/upgrade-plan/constants.tsx:
```diff
@@ -49,23 +49,26 @@ export const UpgradePlanHostingTestimonials = [
+} = {
+	'higher-speed': {
+		title: translate( 'Higher speed' ),
+		description: translate( '30% faster.' ) + '*',
```
See the base for this calc: p1716304442708379/1715972644.965929-slack-C02NWDZBL0H (78.33-47.95)

renatho commented on client/blocks/importer/wordpress/upgrade-plan/constants.tsx:
```diff
@@ -49,23 +49,26 @@ export const UpgradePlanHostingTestimonials = [
-		id: 'higher-availability',
+	'faster-response': {
+		title: translate( 'Faster response' ),
+		description: translate( '98% of sites on WordPress.com have a fast response.' ),
```
The base of this calc is a round from 97.8% (April 2024 data - https://cwvtech.report/)

renatho committed:
Update with number from April 2024

renatho commented on client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-hosting-details.tsx:
```diff
@@ -77,11 +77,16 @@ export const UpgradePlanHostingDetails = () => {
 					<p className="import__upgrade-plan-hosting-details-header-subtext">
-						{ translate( 'Check our performance, compared to the average WordPress host' ) }
+						{ translate(
+							'We offer a proven 24% boost* in Core Web Vitals—endorsed by data from Google.'
```
The base for this calc is the CWM from April 2024. (62.1-38.31)

See more: pdxWSz-1pk-p2

renatho committed:
Update JSON with April data and add WordPress technology

renatho committed:
Change asterisk position to improve readability

renatho committed:
Rename JSON

renatho committed:
Make calcs dynamic

renatho commented on client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-hosting-details.tsx:
```diff
@@ -71,17 +72,30 @@ export const UpgradePlanHostingDetails = () => {
 		);
 	}
 
+	const boostPercentage = Math.round(
```
See more: pdxWSz-1pk-p2

renatho commented on client/blocks/importer/wordpress/upgrade-plan/constants.tsx:
```diff
@@ -49,23 +50,46 @@ export const UpgradePlanHostingTestimonials = [
+	'faster-response': {
+		title: translate( 'Faster response' ),
+		description: translate(
+			'%(fasterResponsePercentage)d% of sites on WordPress.com have a fast response.',
```
Use the number directly, since the difference compared to the average wouldn't be relevant.

renatho commented on client/blocks/importer/wordpress/upgrade-plan/constants.tsx:
```diff
@@ -49,23 +50,46 @@ export const UpgradePlanHostingTestimonials = [
+		description:
+			translate( '%(higherSpeedPercentage)d 30% faster.', {
+				args: { higherSpeedPercentage },
+			} ) + '*',
```
The '*' here and the following one are not part of the translation string because it's more a visual element than the text itself, and they can be changed to something else depending on the page presentation.

renatho committed:
Move calcs from constants to a hook file

renatho commented on client/blocks/importer/wordpress/upgrade-plan/use-default-hosting-details.tsx:
```diff
@@ -0,0 +1,53 @@
+
+const higherAvailabilityPercentage = 3;
+
+export const useDefaultHostingDetails = (): DefaultHostingDetails => {
```
I moved it from the constants file because this has some calcs and doesn't seem to fit well in the constants. Also, in the future, all the data from the JSON will probably be dynamically fetched.

I also took the opportunity to change back to the `useTranslate` hook, since [it seems to fix some issues](https://github.com/Automattic/wp-calypso/pull/89857).

renatho committed:
Escape percentage signals

renatho committed:
Refactor variables and strings

renatho committed:
Update higher availability with real data

renatho committed:
Update metrics

renatho commented on client/blocks/importer/wordpress/upgrade-plan/use-default-hosting-details.tsx:
```diff
@@ -0,0 +1,58 @@
+const fasterResponsePercentage = Math.round( cwvtechReportJson[ 'WordPress.com' ].goodFID * 100 );
+
+const wpcomMinutesDowntime = 0;
+const otherHostsAverageDowntime = 63;
```
See the the calc in pdt2If-1lX-p2#comment-978

renatho committed:
Update boost copy to one that makes more sense based on the data

renatho committed:
Update metrics copy

renatho committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into update/import-metrics-fallback

renatho committed:
Update to round

renatho commented on client/blocks/importer/wordpress/upgrade-plan/hooks/use-get-upgrade-plan-hosting-details-list.tsx:
```diff
@@ -1,67 +1,47 @@
-		};
-		newDefaultHostingDetails = newDefaultHostingDetails.filter(
-			( { id } ) => id !== 'increased-speed'
+		const goodLCPercentage = Math.round( 100 * cwvtechReportJson[ 'WordPress.com' ].goodLCP );
```
Before, it was `Math.floor` to avoid saying that we have 100% of goodLCP when it was 99.9...%. But after updating the JSON it's less than that, so I changed it back to `Math.round`.

renatho replied:
Reverted since the new numbers were wrong. See https://github.com/Automattic/wp-calypso/pull/90966#discussion_r1613652549

renatho commented on client/blocks/importer/wordpress/upgrade-plan/cwvtech-report.json:
```diff
@@ -0,0 +1,16 @@
+{
```
@agrullon95, if you have the opportunity to check if it's correct, I'd appreciate it. Because the numbers are a little different from the original one.

I got it from [this report](https://lookerstudio.google.com/u/0/reporting/55bc8fad-44c2-4280-aa0b-5f3f0cd3d2be/page/M6ZPC?params=%7B%22df44%22:%22include%25EE%2580%25800%25EE%2580%2580IN%25EE%2580%2580WordPress%25EE%2580%2580WordPress.com%22%7D), adding the fields CWV, LCP, CLS, FID and exporting. And I used the April data.

donnapep replied:
These numbers seem to have changed quite a bit since March. Curious if you tried to match the same numbers that were captured for March to confirm that we used the same configuration in the report? From there, it would just be a matter of changing the dates from March to April. This assumes that the proper configuration was used initially, of course. 🙂

renatho replied:
They didn't change too much from March based on the report I'm getting.

<img width="887" alt="Screenshot 2024-05-24 at 11 56 32" src="https://github.com/Automattic/wp-calypso/assets/876340/1d8ee9d6-3670-4887-92ce-69b0ca654a74">

But they have a bigger difference compared to the previous JSON we had.

donnapep replied:
I did this as an exercise myself. I was able to get March's numbers for WP.com, but not the ones you have here for April.

Here was my configuration:
![Screenshot 2024-05-24 at 10 59 50 AM](https://github.com/Automattic/wp-calypso/assets/1190420/48be5e0f-45cc-4a61-9b8d-6a56587cac41)

renatho replied:
Ahhh! The difference is probably the client! Now I see that the default one for me is "Mobile" and on your screen I see "Desktop".

renatho replied:
Numbers updated with the Desktop client now. https://github.com/Automattic/wp-calypso/pull/90966/commits/c9d0a4712be58d4a1e1d380e2b3f1b78ddee915e

I also needed to change it [back](https://github.com/Automattic/wp-calypso/pull/90966#discussion_r1610742625) to avoid cases with "100% of the sites".

renatho committed:
Remove style not used anymore

renatho commented on client/blocks/importer/wordpress/upgrade-plan/use-default-hosting-details.tsx:
```diff
@@ -0,0 +1,58 @@
+export const useDefaultHostingDetails = (): DefaultHostingDetails => {
+	const translate = useTranslate();
+
+	return {
```
Refactor it from array to object, as suggested in https://github.com/Automattic/wp-calypso/pull/90717#discussion_r1603326181.

donnapep commented:
I ran some of the copy through ChatGPT as it didn't sound quite right but I was having trouble improving readability. Let me know what you think about these:

> Google data shows that 24% more WordPress.com sites have good Core Web Vitals as compared to other WordPress hosts.

> Higher speed
> WordPress.com has 30% more sites that are faster than those on other WordPress hosts.

For this one:

> 98% of sites on WordPress.com respond fast, in accordance with Google recommendations.

I have no sense of what "fast" is so it feels sorta meaningless. Could we replace "fast" with whatever Google's definition of fast is? Something like:

> 98% of sites on WordPress.com respond within x seconds on the first interaction.

> Higher availability
> WordPress.com has zero downtime, compared to 63 minutes per month for other WordPress hosts.

As a user, I'd be pretty skeptical of the claim that we have no downtime. I'm also used to seeing the inverse; that is, an uptime measurement  (e.g. 99.99% uptime) as opposed to downtime. WDYT about comparing uptime instead of downtime?

renatho commented:
Thank you for your suggestions, @donnapep!

I applied them to iterate and took new screenshots. It seems the texts are big, and maybe not align very well with what was discussed here p1716389806691159/1715972644.965929-slack-C02NWDZBL0H? Maybe too much specific? WDYT of the current version?

<img width="349" alt="Screenshot 2024-05-23 at 19 17 51" src="https://github.com/Automattic/wp-calypso/assets/876340/d8d3e558-ddf1-4476-8996-8368398b5f62">

<img width="346" alt="Screenshot 2024-05-23 at 19 13 48" src="https://github.com/Automattic/wp-calypso/assets/876340/d0b91f94-9775-4d00-a3c2-883f7443610f">

---

> I have no sense of what "fast" is so it feels sorta meaningless. Could we replace "fast" with whatever Google's definition of fast is? Something like:
> 
> Same question for Higher speed now that I think about it.

To make it more clear here in the PR, "Higher Speed" is about the [LCP](https://web.dev/articles/lcp):

> LCP reports the render time of the largest image or text block visible in the viewport, relative to when the user first navigated to the page.

And "Faster Response" is about the [FID](https://web.dev/articles/fid):

> FID measures the time from when a user first interacts with a page (that is, when they click a link, tap on a button, or use a custom, JavaScript-powered control) to the time when the browser is actually able to begin processing event handlers in response to that interaction.

---

Same question for Higher speed now that I think about it.

I tried to apply the same idea of the other text here:

> 98% of sites on WordPress.com renders the largest image or text block visible in the viewport within 2500 milliseconds, in accordance with Google recommendations.

---

> As a user, I'd be pretty skeptical of the claim that we have no downtime. I'm also used to seeing the inverse; that is, an uptime measurement (e.g. 99.99% uptime) as opposed to downtime. WDYT about comparing uptime instead of downtime?

In this case, we'd say 99.99% against 99.85%. It seems more relevant for me if we have it in downtime minutes. Make sense for you when you see these numbers?

renatho committed:
Tweak the copy

donnapep commented:
_Higher speed_

Oh yes, higher speed is much too verbose. 😅  I don't think we need such a technical explanation, but I'm struggling to make it sound good. So maybe for now we can just keep "fast" for that one.

Would this still be an accurate statement?

> WordPress.com has 30% more sites that display faster than those on other WordPress hosts.

_Faster response_

Do we need the "in accordance with Google recommendations" bit? The default ones don't have that disclaimer, so I'm just trying to figure out where it came from. Is it to make the metric sound more official / reputable? I also converted to seconds since that's an easier measurement to grok.

> 98% of sites on WordPress.com respond within 0.1 seconds on the first interaction.

_Uptime_

> In this case, we'd say 99.99% against 99.85%. It seems more relevant for me if we have it in downtime minutes. Make sense for you when you see these numbers?

Yes, I think that comparing uptime percentages makes more sense (unless there's been conversation elsewhere to use downtime?). The 0 minutes would make me not trust WordPress.com since it's impossible to achieve on an ongoing basis. We also talk about uptime on the [move landing page](https://wordpress.com/move/). So perhaps this?

> WordPress.com boasts a 99.99% uptime, compared to 99.85% uptime for other WordPress hosts.

Do these suggestions make sense given the discussions that you've seen in P2 comments and posts for what we're trying to communicate?

renatho committed:
Tweak the copy

renatho commented:
@donnapep, I tweaked the copy based on the last suggestions. https://github.com/Automattic/wp-calypso/pull/90966/commits/36dac8bb3e5a79adf02fea375f1632754740f93a

<img width="343" alt="Screenshot 2024-05-24 at 10 51 22" src="https://github.com/Automattic/wp-calypso/assets/876340/cdecbf65-9ce5-4300-8aa1-5dcd5a3458a2">

---

> Would this still be an accurate statement?
> 
> > WordPress.com has 30% more sites that display faster than those on other WordPress hosts.

Thinking a little more, I changed it to "WordPress.com has 30% more sites that display fast compared to other WordPress hosts.". WDTY?

The logic behind that is that we have 78.3% of sites with good LCP (2.5 seconds or less) on WPCOM against 47.9% of sites with good LCP in other hosts.

So I think we could have more (or less) faster sites than WordPress hosts, but here we're just comparing the percentage of sites that is considered fast (with a good LCP in this case).

If you think the "faster than" still makes sense with that, let me know and I'll change it back.

> Do we need the "in accordance with Google recommendations" bit? The default ones don't have that disclaimer, so I'm just trying to figure out where it came from.

It came when I updated the original copy from "30% faster" for the first time. The reason I added was to justify what is "fast". When you used "faster" it solved this, and maybe I broke it again changing back to "fast". Anyway, I think the last version is not super bad. WDYT?

> Yes, I think that comparing uptime percentages makes more sense (unless there's been conversation elsewhere to use downtime?). The 0 minutes would make me not trust WordPress.com since it's impossible to achieve on an ongoing basis. We also talk about uptime on the [move landing page](https://wordpress.com/move/). So perhaps this?

Good point! It makes sense to keep it aligned with what we already present! =)

> Do these suggestions make sense given the discussions that you've seen in P2 comments and posts for what we're trying to communicate?

I hope so! I know that design wanted to have less text, but I think it's very hard to achieve because the information is not so simple.

renatho commented on client/blocks/importer/wordpress/upgrade-plan/use-default-hosting-details.tsx:
```diff
@@ -0,0 +1,61 @@
+const fastResponsePercentage = Math.round( cwvtechReportJson[ 'WordPress.com' ].goodFID * 100 );
+
+const wpcomPercentageUptime = '99.99';
+const otherHostsPercentageUptime = '99.85';
```
See the the calc in pdt2If-1lX-p2#comment-978

renatho committed:
Update numbers filtering with Desktop client

donnapep commented:
> Thinking a little more, I changed it to "WordPress.com has 30% more sites that display fast compared to other WordPress hosts.". WDTY?

I get what you're saying, though it still reads awkwardly. How about this?

> WordPress.com has 30% more sites that load quickly compared to other WordPress hosts.

All the other metrics look good. 😃

renatho committed:
Tweak copy

renatho commented:
> I get what you're saying, though it still reads awkwardly. How about this?

Looks good to me! Updated in https://github.com/Automattic/wp-calypso/pull/90966/commits/1d0c70d90d8ce83139e2cdbd1e1151c3ea4613c5
Thank you for all the help with it, Donna! 🫶

<img width="347" alt="Screenshot 2024-05-24 at 12 31 15" src="https://github.com/Automattic/wp-calypso/assets/876340/38e0166e-37ca-4855-9df2-aacd73fd3009">

<img width="348" alt="Screenshot 2024-05-24 at 12 32 10" src="https://github.com/Automattic/wp-calypso/assets/876340/165874c4-511d-421b-9a83-e6855a90667e">

donnapep reviewed (approved):
I think it's as good as it's going to get! 😄 Thanks for your persistence through all the tweaks and re-tweaks. 👏🏻

renatho commented:
Since it had many string changes, I added the String Freeze label. I'll wait for it before deploying.
Meanwhile, if someone has any other insights, we can iterate. ;)

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/13555535

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @renatho for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/13555535).
